src/types.rs

```diff
@@ -634,11 +647,64 @@ impl Event {
         // part of the URI and we don't need any additional information (the "cluster" argument is
         // ignored).
         let self_activity = self.data.activity.as_ref();
-        GlobalId::new_with_cluster_unchecked(
+        let id = match self.activity_type() {
+            ActivityType::Space(SpaceActivity::Created) => self.room_id_of_space_created_event()?,
+            ActivityType::Space(
+                SpaceActivity::Changed | SpaceActivity::Joined | SpaceActivity::Left,
+            ) => self
+                .data
+                .activity
+                .clone()
+                .and_then(|a| a.target)
+                .and_then(|t| t.global_id)
+                .ok_or(crate::error::ErrorKind::Api(
+                    "Missing global ID in space changed event",
+                ))?,
+            _ => self_activity.map_or_else(|| self.id.clone(), |a| a.id.clone()),
+        };
+        Ok(GlobalId::new_with_cluster_unchecked(
             self.activity_type().into(),
-            self_activity.map_or_else(|| self.id.clone(), |a| a.id.clone()),
+            id,
             None,
-        )
+        ))
+    }
+
+    /// Get the UUID of the room the Space created event corresponds to.
+    /// This is a workaround for a bug in the API, where the UUID returned in the event is not correct.
+    ///
+    /// # Panics
+    ///
+    /// Will panic if the event is not `Space::Created` or if activity is not set.
+    fn room_id_of_space_created_event(&self) -> Result<String, crate::error::Error> {
+        assert_eq!(
+            self.activity_type(),
+            ActivityType::Space(SpaceActivity::Created),
+            "Expected space created event, got {:?}",
+            self.activity_type()
+        );
+        let id = self
+            .data
+            .activity
+            .clone()
+            .ok_or(crate::error::ErrorKind::Api(
+                "Missing activity in space created event",
+            ))?
+            .id;
+        // If the id is not a UUID, assume it is already a correct global ID.
+        // This could not be tested though as the API only returns UUID for now.
+        if Uuid::parse_str(&id).is_err() {
+            return Ok(self.id.clone());
+        }
+        // API weirdness... the event contains an id that is close to the room id,
+        // but it is not the same. It differs from the room id by one character,
+        // always by a value of 2.
+        let mut uuid = id;
+        if uuid.as_bytes()[7] == b'2' {
+            uuid.replace_range(7..8, "0");
+        } else {
+            panic!("Space created uuid {uuid} could not be not patched");
```

Comment:
I really do not like a panic, I would rather return an Error since we already are returning a Result

Reply:
something like 
```rust
crate::error::ErrorKind::Api(
                "Space created uuid {uuid} could not be not patched",
            )
            ```

Reply:
oh yes, that one slipped through the cracks from a previous version! I'll remove it